src/pages/register-a-beacon/application-complete.tsx

```diff
@@ -90,16 +90,22 @@ const ApplicationCompleteYourBeaconRegistryAccount: FunctionComponent =
 export const getServerSideProps: GetServerSideProps = withContainer(
   async (context: BeaconsGetServerSidePropsContext) => {
     /* Retrieve injected use case(s) */
-    const { submitRegistration, getAccountHolderId } = context.container;
+    const { submitRegistration, getAccountHolderId, getSession } =
+      context.container;
 
     /* Page logic */
     if (!verifyFormSubmissionCookieIsSet(context))
       return redirectUserTo(PageURLs.start);
 
     try {
+      const accountHolderEmail: string = await (
+        await getSession(context)
+      ).user.email;
+
       const result = await submitRegistration(
         retrieveUserFormSubmissionId(context),
-        await getAccountHolderId(context)
+        await getAccountHolderId(context),
+        accountHolderEmail
       );
```

Comment:
Isn't `getSession` an implementation detail, not a use case?  It's more like a concrete `sessionGateway`?

If we couple code to the structure of `session` (as here at line 103, `.user.email)`, it'll be difficult to change in future and harder to grok because the use cases are at different levels of abstraction. If the structure of `session` is defined by MSAL (?), this is unnecessary coupling between the `application-complete` page and B2C?

Is this also the right place to put this logic?  We created the use cases (like `submitRegistration`) to encapsulate business rules.  Adding another parameter to inject an email into `submitRegistration` as well feels like a leaking abstraction, in the sense that this page is having to tell `submitRegistration` how to do its job.

Really sorry Matt but I think this is the wrong place to make this change.  The `submitRegistration` use case could be extended to pull the `accountHolderApiGateway` from the app container and use that to do logic of identifying the account holder and sending them an email?

Sorry to be a purist :/ Happy to 🍐  on this.

Reply:
@zackads Totally agree on the above, I wanted to bring out injecting the email into the `sendConfirmationEmail` use case but went one level too high up into the `application-complete` page!

I have updated this PR to extract the email for the account holder from the `accountHolderApiGateway` and inject that into the `sendConfirmationEmail` use case.

Please let me know if there's anything else that needs changing on the PR